components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/APIMgtLatencySynapseHandler.java

```diff
@@ -34,14 +34,18 @@
 
     private TracingTracer tracer;
 
-    public APIMgtLatencySynapseHandler() {
-        tracer = ServiceReferenceHolder.getInstance().getTracingService()
-                .buildTracer(APIMgtGatewayConstants.SERVICE_NAME);
-    }
-
     @Override
     public boolean handleRequestInFlow(MessageContext messageContext) {
+
         if (Util.tracingEnabled()) {
+            if (tracer == null) {
```

Comment:
While this will work in most of the cases, wouldn't it cause a problem when an OSGI module unloads/re-loads? Can we trigger the action of building tracer from setTracingService method on ServiceComponent?